andela/coding-geeks-ah-backend
# #167484589 implemented social login

#### What does this PR do?
It implements the approach of being able to login with social network platforms such as Google, Twitter, and Facebook.

#### Description of Task to be done?
1. Adding routes for social login with Google, Twitter, or Facebook.
2. Adding callback routes for storing profile information from social platforms.
3. Adding tests for the social login routes and controllers.

#### Screenshots
<img width="1319" alt="Screen Shot 2019-08-17 at 20 41 50" src="https://user-images.githubusercontent.com/34879554/63216043-8aee2f80-c12f-11e9-8a89-65640ce9986d.png">
<img width="1291" alt="Screen Shot 2019-08-17 at 20 42 07" src="https://user-images.githubusercontent.com/34879554/63216045-90e41080-c12f-11e9-8758-213486c805d7.png">
<img width="1287" alt="Screen Shot 2019-08-17 at 20 42 19" src="https://user-images.githubusercontent.com/34879554/63216046-92153d80-c12f-11e9-8366-80362c16beff.png">

#### How should this be manually tested?
1. Clone this branch.
2. Don **npm install**.
3. Add values to these variables in your **.env** file:
FACEBOOK_APP_ID=
FACEBOOK_APP_SECRET=
GOOGLE_APP_ID=
GOOGLE_APP_SECRET=
TWITTER_APP_KEY=
TWITTER_APP_SECRET=
TWITTER_TOKEN=
TWITTER_TOKEN_SECRET=
4. Log into your Gmail, Twitter, and Facebook accounts for being able to test the social login feature.
5. Then, visit corresponding social login routes for each platform, Google, Twitter, and Facebook:
localhost:3000/api/v1/auth/google, localhost:3000/api/v1/auth/twitter/, and localhost:3000/api/v1/auth/facebook, with your browser.
6. Authorize **Authors Haven** app to access to your account.
7. Check if you get a message telling you that you have logged in with your preferred platform.
